Firestore/Example/Tests/Model/FSTMutationTests.mm

```diff
@@ -130,11 +140,60 @@ - (void)testAppliesLocalTransformsToDocuments {
   XCTAssertEqualObjects(transformedDoc, expectedDoc);
 }
 
+// NOTE: This is more a test of FSTUserDataConverter code than FSTMutation code but we don't have
+// unit tests for it currently. We could consider removing this test once we have integration tests.
+- (void)testCreateArrayUnionTransform {
+  FSTTransformMutation *transform = FSTTestTransformMutation(@"collection/key", @{
+    @"foo" : [FIRFieldValue fieldValueForArrayUnion:@[ @"tag" ]],
+    @"bar.baz" : [FIRFieldValue fieldValueForArrayUnion:@[ @YES, @[ @1, @2 ], @{@"a" : @"b"} ]]
+  });
+  XCTAssertEqual(transform.fieldTransforms.size(), 2);
+
+  const FieldTransform &first = transform.fieldTransforms[0];
+  XCTAssertEqual(first.path(), FieldPath({"foo"}));
+  {
+    std::vector<FSTFieldValue *> expectedElements;
+    expectedElements.push_back(FSTTestFieldValue(@"tag"));
+    const ArrayTransform expected(TransformOperation::Type::ArrayUnion, expectedElements);
```

Comment:
The extra `const` in here serves no real purpose given that we're already making these things immutable.

Reply:
Done.